### PR TITLE
Make Series.quantile compat with pandas

### PR DESCRIPTION
**Pull request by sinhrks**
``Series.quantile`` has some inconsistencies with pandas.

```
import pandas as pd
import dask.dataframe as dd

s = pd.Series([1, 2, 3, 4, 5])
ds = dd.from_pandas(s, 2)
```

#### 1. ``pandas`` quantile can accept scalar, but dask doesn't.

```
s.quantile(0.5)
3.0

ds.quantile(0.5)
# TypeError: object of type 'numpy.float64' has no len()
```

#### 2. ``pandas`` quantiles with list-like have percentile as index, but dask doesn't. 

Also, dask's division must be lower / upper percentiles, e.g. (0.2, 0.8), not keep original divisions.

```
s.quantile([0.2, 0.8])
# 0.2    1.8
# 0.8    4.2
# dtype: float64

ds.quantile([0.2, 0.8]).compute()
# 0    1.4
# 1    4.8
# dtype: float64

ds.quantile([0.2, 0.8])
# dd.Series<quantiles-3-74ec7e1bfb67fad191323f3109f0da62, divisions=(0, 4)>
```

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -487,6 +484,13 @@ def _args(self):
         return (self.dask, self._name, self.name, self.divisions)
 
     @property
+    def _partition_type_sliced(self):
+        # return scalar
+        return lambda values, *args: values.item()
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Could this be the following:

    _partition_type_sliced = pd.Series.item

**Comment by mrocklin** [comment]
+1

File: dask/dataframe/core.py (hunk 1 above)
Comment (sinhrks) [inline dask/dataframe/core.py]: Thanks, fixed to ``lambda v: df._partition_type(v).item()``, and removed unnecessary generalization as it only supports ``Series``.

**Comment by cowlicks** [comment]
Looks good, thanks @sinhrks

**Pull request merged**